apps/ejabberd/src/mod_aws_sns_defaults.erl

```diff
@@ -0,0 +1,48 @@
+%%%-------------------------------------------------------------------
+%%% @author Rafal Slota
+%%% @copyright (C) 2017 Erlang Solutions Ltd.
+%%% This software is released under the Apache License, Version 2.0
+%%% cited in 'LICENSE.txt'.
+%%% @end
+%%%-------------------------------------------------------------------
+%%% @doc
+%%% Plugin module defining default custom behaviour for AWS SNS notifications.
+%%% @end
+%%%-------------------------------------------------------------------
+-module(mod_aws_sns_defaults).
+-author("Rafal Slota").
+
+-behavior(mod_aws_sns).
+
+-include_lib("ejabberd/include/jlib.hrl").
+
+%%%===================================================================
+%%% Exports
+%%%===================================================================
+
+%% Callbacks
+-export([user_guid/1, message_attributes/3, message_attributes/5]).
+
+%% -------------------------------------------------------------------
+%% @doc Returns user's GUID for SNS notification based on his JID.
+%% -------------------------------------------------------------------
+-spec user_guid(UserJID :: ejabberd:jid()) -> mod_aws_sns:user_guid().
+user_guid(#jid{} = UserJID) ->
+    jid:to_binary(jid:to_lower(jid:to_bare(UserJID))).
+
+%% -------------------------------------------------------------------
+%% @doc Returns SNS Message Attributes for presence change notification.
+%% -------------------------------------------------------------------
+-spec message_attributes(TopicARN :: mod_aws_sns:topic_arn(), UserJID :: ejabberd:jid(),
+                         IsOnline :: boolean()) -> mod_aws_sns:attributes().
+message_attributes(_TopicARN, _UserJID, _IsOnline) ->
+    #{}.
```

Comment:
This function is described as returning mod_aws_sns:attributes(), but it returns a map - which happens to be the same, but doesn't have to. I'd think of making mod_aws_sns:attributes() an opaque type, but it might be an overkill, so you better discuss it with someone else.